### ISSUE TITLE
Ability to sign http request

### ISSUE THREAD

**Issue by linvi** [linvi/tweetinvi#220]
`TwitterAccessor` could have a new feature that allow the developer to sign a request to execute on any of the Twitter public endpoints.

This method will return all the headers (and potentially additional information) required for a user to execute the query client side but yet using the signature created by Tweetinvi.

**Comment by linvi** [comment]
`ITwitterQuery` now has a `AuthorizationHeader` property that can be used.
`TwitterAccessor.GenerateTwitterRequestParameters` allow developers to get all the information required to run a query from another server.

**Comment by JoshKeegan** [comment]
Really cool that this has been added :)
I've been having a play with it & put together a demo at https://github.com/JoshKeegan/TweetinviRequestSigner. 

Were you intending for `TwitterAccessor.GenerateTwitterRequestParameters` to be used server-side to sign the request? If so, what's the advantage over using `OAuthWebRequestGenerator` directly, as in https://github.com/JoshKeegan/TweetinviRequestSigner/blob/master/TweetinviRequestSigner/Controllers/RequestController.cs ?
I looked at the new method, but it didn't seem necessary. I'll no doubt have missed something here!
